daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilParseXMLReader.scala

```diff
@@ -144,7 +142,13 @@ class DaffodilParseXMLReader(dp: DataProcessor) extends DFDL.DaffodilParseXMLRea
   }
 
   def parse(isdis: InputSourceDataInputStream): Unit = {
-    val sio = createSAXInfosetOutputter(this)
+    val sio = createSAXInfosetOutputter(this,
```

Comment:
Unrelated, but I noticed above the setProperty function just calls value.asInstanceOf. If value isn't an instance of the right thing, I think that will throw a ClassCastException? Should that first check if it's the right instance, and then throw SAXNotRecognizedException or SAXNotSupportedException?

Reply:
So as opposed to calling asinstanceOf, then catching ClassCastExpression and throwing the SaxNotSupported, We check if the value that instance, thereby avoiding it throwing the ClassCatException all together? Something like:

```scala
        case XMLUtils.DAFFODIL_SAX_URN_BLOBDIRECTORY => {
          if (!value.isInstanceOf[Path]) {
            throw new SAXNotSupportedException("Unsupported value for property: " + name + ". Expected Path.")
          } else {
            saxBlobDirectoryPropertyValue = value.asInstanceOf[Path]
          }
        }
```

Reply:
Ah, I missed the ClassCastException, what you already have is prefectly reasonable. Ignore this comment.